src/pynxtools/testing/nexus_conversion.py

```diff
@@ -51,8 +53,8 @@ def __init__(self, nxdl, reader_name, files_or_dir, tmp_path, caplog) -> None:
         files_or_dir : str
             List of input files or full path string to the example data directory that contains all the files
             required for running the data conversion through the reader.
-        ref_nexus_file : str
-            Full path string to the reference NeXus file generated from the same
+        ref_log_file : str
+            Full path string to the reference log file generated from the same
```

Comment:
This is redundant. The log files will come in place of `nexus_file` in `files_or_dir` from the plugins.

Reply:
I disagree. What if the reader can support two different appdefs, then I need two reference log files for the same data set. I would not like to store the same data twice in two folders, but rather just two log files in the same `files_or_dir`. See [here ](https://github.com/FAIRmat-NFDI/pynxtools-xps/pull/69/files#diff-030d39f7d85e7dd25282dad362350a7a06cd685af9fa186d8ecf4e83c8f6028b)for an example in pynxtools-xps (supporting NXmpes and NXxps).

I guess we could just pass the `ref_log_file ` as part of a list of files to `files_or_dir`, but I would prefer to have the option to pass it directly to the function.

Reply:
I can not think such `generic_reader` that will be fit for multiple appdefs. Even it supports multiple `appdef` then `eln`, `config_file` or even raw must be different. 

However, `ref_log_file` can also be added in the list of the files, not in the `dir`. In that case, using multiple `appdef` for the same reader will also be applicable.

The reason, I wanted to reduce the unnecessary function `arg`, which keeps the `test framework` easy to use and intuitive to external users.
But, if you are optimistic about keeping the `ref_log_file` no problem it can be kept.

Reply:
Look at the link, the data file, `eln`, `config`, and so on are the same for `NXxps` and `NXmpes` (because `NXxps` extends `NXmpes`). For me, having it as an argument is fine, if not needed, one can simply set it to None. As a compromise, I set to None by default, so if you don't give any value for `ref_log_file`, you can simply omit it and the log file is directly parsed from `files_or_dir`.